client/src/globals.ts

```diff
@@ -137,3 +153,19 @@ if ((window as any).CELLXGENE && (window as any).CELLXGENE.API) {
 }
 
 export const API = _API;
+
+/*
+ Update the base API URL for the current dataset using the current origin and pathname. Noop for localhost as switching
+ between datasets is not enabled for local environments.
+ */
+export function updateApiPrefix(): void {
+  if (typeof window === "undefined") {
+    throw new Error("Unable to set API route.");
+  }
+  const {
+    location: { host, pathname, protocol },
+  } = window;
+  if (!origin.includes("localhost")) {
+    API.prefix = `${protocol}//api.${host}/cellxgene${pathname}api/`;
```

Comment:
This is already set up by an inline script. Why do we need this function as well?

Reply:
Thanks Emanuele, Timmy. I'll take a look at updating this to use a regex.

Reply:
@tihuan, @ebezzi, @MDunitz, I have updated [updateApiPrefix()](https://github.com/chanzuckerberg/single-cell-explorer/pull/111/files#diff-ec42b63c529ac770308956529c4f271f479390904aa3d7a59d8dc284abfcac59R166) to use  the [regex](https://github.com/chanzuckerberg/single-cell-explorer/pull/111/files#diff-ec42b63c529ac770308956529c4f271f479390904aa3d7a59d8dc284abfcac59R16) `/(?<=\/)\w+\/[^/]+\/(?=api)/`.